### PR TITLE
[spec] Add missing range checks on memory limits

### PR DESCRIPTION
**Pull request by rossberg**
Address #861. Also tweak interpreter to match new rule structure.

### CODE REVIEW

File: document/core/valid/types.rst [hunk 1]
```diff
@@ -13,20 +13,30 @@ However, restrictions apply to :ref:`function types <syntax-functype>` as well a
...
+
+  * Its value must not be larger than :math:`k`.
+
+  * Its value must not be smaller than :math:`n`.
+
+* Then the limit is valid within range :math:`k`.
 
 .. math::
    \frac{
+     n < k
```
Comment (binji) [inline document/core/valid/types.rst]: \leq?

File: document/core/valid/types.rst [hunk 2]
```diff
@@ -13,20 +13,30 @@ However, restrictions apply to :ref:`function types <syntax-functype>` as well a
...
+     \qquad
+     (m < k)^?
```
Comment (binji) [inline document/core/valid/types.rst]: same here?

File: document/core/valid/types.rst (hunk 1 above)
Comment (rossberg) [reply document/core/valid/types.rst]: Fixed.

File: document/core/valid/types.rst (hunk 2 above)
Comment (rossberg) [reply document/core/valid/types.rst]: Fixed.

**Pull request merged**